### PR TITLE
Add documentation for the RuntimeCleaner and Crawdad daemonset

### PR DESCRIPTION
**Pull request by Brian-McM**
Product Version(s):

Issue:

Link to docs preview:

SME review:
- [ ] An SME has approved this change. 

DOCS review:
- [ ] A member of the docs team has approved this change.

Additional information:

### CODE REVIEW

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+---
+
+# Configuring the Runtime View
+
+**Enable Runtime View**
+
+When Runtime View is enabled in Scan Settings, {{prodname}} deploys a Daemonset, named "Crawdad", into your connected Managed Clusters to
+actively monitor pods running in your Kubernetes clusters and gather image information from the node
+they're running on. This information is used to correlate the images your pods are running with the images you
+have upload scan results for.
```
Comment (stevegaossou) [inline calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: Is this supposed to be: 

> have **uploaded** scan results for.

Or: 

> have scan results for.

**Comment by stevegaossou** [comment]
@Brian-McM I noticed the preview version of this PR does not include your changes. It would be worth checking with the docs team to see whether preview works for all PRs or not. 

I think there's some extra configuration you need to at least have the new page included in the side menu bar that's currently missing in this PR.

**Comment by ghost** [comment]
I am going through some test cycles for this feature, and the "Enable Runtime View" toggle was a bit confusing to me. I added a new cluster to my management cluster and I expected to have to enable the feature for this specific cluster, but it was already enabled. Can we change the language to show the customer that the toggle for this enables it across all their clusters? Also, is the score thresholds universal as well? Some language to explain this would be great as well.

**Comment by Brian-McM** [comment]
@stevegaossou there's tweaks we can make to the Makefile target to show this, but currently cloud is pinned to show only 3.15 changes (which is not where this belongs). They haven't made the 3.16 section yet, but I'll check to see if there's any precedent on setting this for the PR so it's viewable.

**Comment by ctauchen** [comment]
@Brian-McM Some context would be helpful. Can you add some information so the docs team know what we're reviewing? Thanks!

**Comment by Brian-McM** [comment]
Need to add information about setting the CR socket path, for rancher (or anything using cri-dockerd) it's `/var/run/cri-dockerd.sock`, for kubernetes versions before 1.23 (or anything else using dockershim) it's `/var/run/dockershim.sock`

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+---
+description: Use the Runtime View to find vulnerabilities in your cluster.
+---
+
+# Configuring the Runtime View
+
+**Enable Runtime View**
```
Comment (Brian-McM) [inline calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: Need to show / describe how to enable the run time view, not just what happens when you do.

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+---
+description: Use the Runtime View to find vulnerabilities in your cluster.
+---
+
+# Configuring the Runtime View
+
+:::warning
+
+The Runtime View requires that all the connected Managed Clusters are running Kubernetes using a CRI for container management.
```
Comment (Brian-McM) [inline calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: Maybe state kubernetes versions and link to something explaining what the CRI is.

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx [hunk 4]
```diff
@@ -0,0 +1,63 @@
+---
+description: Use the Runtime View to find vulnerabilities in your cluster.
```
Comment (ghost) [inline calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: I am wondering if this better stated as something along the lines of "Use the Runtime View to see scan results of images in running pods within your cluster"

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+---
+
+# Configuring the Runtime View
+
+## Big picture
+
+View pods running vulnerability images for all your Managed Clusters.
```
Comment (ghost) [inline calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: Is this better stated as: "See scan results of images in running pods within your cluster"?

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+## Value
+
+Runtime View collects image information from all the pods running in all of your connected Managed Clusters. This information is linked to uploaded image scan results so you can see if you clusters are running vulnerable images.
```
Comment (ghost) [inline calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: Typo: to see if _your_

Comment (ghost) [inline calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: I am wondering if we should mention the Tigera image scanner here? I was thinking something along the lines of: The runtime view shows images running in pods and their associated metadata. Vulnerability information, if available either with the Tigera image scanner or the in cluster scanner, will be shown here.

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+
+## How does the Runtime View collect image information from my Managed Clusters
+
+When Runtime View is enabled, {{prodname}} deploys a Daemonset, named "Crawdad", into your connected Managed Clusters to
```
Comment (ghost) [inline calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: Is it supposed to be: When _the_?

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+
+:::
+
+## Before you begin
+
+**Required**
+
+Ensure that all Managed Clusters connected to your Management Cluster are running a Kubernetes using a CRI for container management.
+
+**Enable Runtime View**
```
Comment (ghost) [inline calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: I think we could say explicitly that this enables it for all the managed clusters.

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+To enable the Runtime View go to the Scan Settings and check the "Enable Runtime View" box. The settings can be
+accessed by clicking the gear on the right side of the page in the Scan Results tab.
+
+By default, this feature is disabled. If you disable the feature after it has been enabled, runtime information that was
```
Comment (ghost) [inline calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: Is this still true? I thought it removed it. I could be incorrect here.

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+already collected will persist.
+
+**Configuring the CRI socket path for Managed Cluster**
+
+Some Kubernetes clusters use different CRI binaries other than the default containerd one, or the containerd CRI socket
+might be located other than the default location. If the Crawdad Pods running in your Managed Cluster don't have the correct
```
Comment (ghost) [inline calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: Is this supposed to say: located _somewhere other than_?

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+location of the CRI socket, they won't be able to gather image information for the pods running in your cluster.
+
+Other than containerd, we know of two other CRI implementations (there may be others):
```
Comment (ghost) [inline calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: I am wondering if we should say that these are examples of other CRI implementations? Other then saying we know of them.

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+1. cri-dockerd, whose socket path is at `/var/run/cri-dockerd.sock`.
+2. dockershim, whose socket path is at `/var/run/dockershim.sock`
+
+To update the CRI socket path for the Crawdad Pods use the following command:
+```bash
+kubectl patch imageassurance default --type='merge' -p '{"spec":{"criSocketPath":"<path-to-cri-socket"}}'
```
Comment (ghost) [inline calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: Typo: missing > character

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+```
+
+We know that some kubernetes platforms or k8s versions use different CRI binaries, here are some examples of where we
+know the CRI socket path is different from the default:
+1. Rancher uses cri-dockerd, so the CRI socket path should be located at `/var/run/cri-dockerd.sock`.
+2. EKS installations with kubernetes version lower than 1.23 use docker shim, located at `/var/run/dockershim.sock`
+3. RKE2 installation use containerd but it's located at `/run/k3s/containerd/containerd.sock`
```
Comment (ghost) [inline calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: Typo: installations / uses

File: calico-cloud/image-assurance/scan-image-registries.mdx [hunk 14]
```diff
@@ -222,6 +222,8 @@ In the Scan Results and Runtime tabs, the **Registry path** field may be blank i
 
 :::
 
+For information configuring the Runtime View please see [Configuring the Runtime View](configuring-the-runtime-view.mdx)
```
Comment (ghost) [inline calico-cloud/image-assurance/scan-image-registries.mdx]: Typo: For information _on_ configuring

File: calico-cloud/reference/installation/api.mdx [hunk 15]
```diff
@@ -5,7 +5,14 @@ description: Installation API reference
...
 import API from '@site/calico-cloud/reference/installation/_api.mdx';
+import IAAPI from '@site/calico-cloud/reference/installation/_ia_api.mdx';
 
 The Kubernetes resources below configure {{prodname}} installation when using the operator. Each resource is responsible for installing and configuring a different subsystem of {{prodname}} during installation. Most options can be modified on a running cluster using `kubectl`.
 
 <API />
+
+## Image Assurance installation reference
+
+The Kuberenetes resources below configure Image Assurance installation when using the operator.
```
Comment (ghost) [inline calico-cloud/reference/installation/api.mdx]: Typo: _configures the_

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx (hunk 7 above)
Comment (Brian-McM) [reply calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: No, this is intentional. It's how the grammar works for product names.

**Review by ctauchen — COMMENTED** [review]
@Brian-McM This structure is much more friendly to our users. Ping me when you've got the doc itself ready to review.

File: calico-cloud/image-assurance/change-the-cri-socket-path.mdx [hunk 16]
```diff
@@ -0,0 +1,41 @@
+---
+description: Change the Runtime View CRI socket path from the default value.
+---
+
+# Change the Runtime View CRI socket path.
+
+## Big picture
+
+Change the CRI socket path in connected Managed Clusters from the path
```
Comment (ghost) [inline calico-cloud/image-assurance/change-the-cri-socket-path.mdx]: Typo: trailing "from the path"

File: calico-cloud/image-assurance/change-the-cri-socket-path.mdx [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+Below is a list of systems that are known to use CRI implementations other than Containerd, or the Containerd socket
+is located some other than the default location:
+1. Rancher uses cri-dockerd, so the CRI socket path should be located at `/var/run/cri-dockerd.sock`.
+2. EKS installations with kubernetes version lower than 1.23 use docker shim, located at `/var/run/dockershim.sock`
+3. RKE2 installation use Containerd but it's located at `/run/k3s/containerd/containerd.sock`
+4. Installations of Kubernetes versions lower than 1.23 might be using `dockershim` by default, located at `/var/run/dockershim.sock`
+
+**Configuring the CRI socket path for Managed Cluster**
+
+To update the CRI socket path for you Managed Cluster use the following command:
```
Comment (ghost) [inline calico-cloud/image-assurance/change-the-cri-socket-path.mdx]: Typo: "your" Manager Cluster

File: calico-cloud/image-assurance/change-the-cri-socket-path.mdx [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+```bash
+kubectl patch imageassurance default --type='merge' -p '{"spec":{"criSocketPath":"<path-to-cri-socket"}}'
```
Comment (ghost) [inline calico-cloud/image-assurance/change-the-cri-socket-path.mdx]: Typo: Missing ">"

File: calico-cloud/image-assurance/change-the-cri-socket-path.mdx [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+```
+
+See [the installation reference](../reference/installation/api.mdx#image-assurance.operator.tigera.io/v1.ImageAssuranceSpec) for more details on the ImageAssurance CR.
```
Comment (stevegaossou) [inline calico-cloud/image-assurance/change-the-cri-socket-path.mdx:41]: @Brian-McM should we call out somewhere that if the daemon set is not deployed correctly, the user can check the tigerastatus for more information?

Comment (Brian-McM) [reply calico-cloud/image-assurance/change-the-cri-socket-path.mdx:41]: I think we'll need to do a follow up PR to re work the runtime view docs. @ctauchen had some ideas about making the overall description of the runtime view better.

Phil had said nobody is using this currently, so we can skip the initial release notes and warnings, so for now we're just going to highlight how to configure it to work if the default configuration doesn't work.

File: calico-cloud/image-assurance/change-the-cri-socket-path.mdx (hunk 16 above)
Comment (Brian-McM) [reply calico-cloud/image-assurance/change-the-cri-socket-path.mdx]: That was supposed to be "from the default path", thanks for the catch.

File: calico-cloud/image-assurance/scan-image-registries.mdx (hunk 14 above)
Comment (Brian-McM) [reply calico-cloud/image-assurance/scan-image-registries.mdx]: Not applicable anymore, restructure the docs.

File: calico-cloud/reference/installation/api.mdx (hunk 15 above)
Comment (Brian-McM) [reply calico-cloud/reference/installation/api.mdx]: changes installation to installations instead

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx (hunk 6 above)
Comment (Brian-McM) [reply calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: Change the structure, N/A

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx (hunk 11 above)
Comment (Brian-McM) [reply calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: Change the structure, N/A

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx (hunk 8 above)
Comment (Brian-McM) [reply calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: Change the structure, N/A

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx (hunk 10 above)
Comment (Brian-McM) [reply calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: Change the structure, N/A

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx (hunk 9 above)
Comment (Brian-McM) [reply calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: No it doesn't remove it.

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx (hunk 6 above)
Comment (Brian-McM) [reply calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: Change the structure, N/A

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx (hunk 5 above)
Comment (Brian-McM) [reply calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: Change the structure, N/A

File: calico-cloud/image-assurance/configuring-the-runtime-view.mdx (hunk 4 above)
Comment (Brian-McM) [reply calico-cloud/image-assurance/configuring-the-runtime-view.mdx]: Change the structure, N/A

**Comment by ctauchen** [comment]
Changes tracked now in https://github.com/tigera/docs/pull/581

**Pull request closed without merging**